src/bsoncxx/private/error_category.cpp

```diff
@@ -137,15 +137,18 @@ const std::error_category& bsoncxx_error_category() {
 
 const std::error_category& libbson_error_category(int domain) {
     switch (domain) {
-        case BSON_ERROR_JSON:
+        case BSON_ERROR_JSON: {
```

Comment:
This was actually a real bug that MSVC told us about. Without braces in each case, the scope of the static category extends to the entire switch statement, but you aren't allowed to jump past the initialization of static.